dbn_upper_bound/python/mputility.py

```diff
@@ -237,13 +237,116 @@ def Ht_AFE_ABC(z, t):
     A = Ht_AFE_A(z, t)
     B = Ht_AFE_B(z, t)
     C = Ht_AFE_C(z, t)
-
+    
     H = A+B+C
     if z.imag == 0:
         return H.real
     else:
         return H
 
+def AB_analysis(z, t):
```

Comment:
What is the goal of AB_analysis and of adjusted_AB_analysis? It isn't being used anywhere else from what I see.

If they aren't generic utilities, then they should be in a separate file.